src/components/Molecules/CtaPrompt/CtaPrompt.component.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file CtaPrompt.component.js
+ * Exports a CTA Prompt component.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import CtaMessage from '../CtaMessage/CtaMessage.component';
+
+const cookiePrefix = 'cta_prompt_';
+
+/**
+ * Component that renders a CTA (Call to action) prompt.
+ */
+export default class CtaPrompt extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    type: CtaMessage.propTypes.type
+  };
+
+  static defaultProps = {
+    children: [],
+    type: null
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      shownMessage: null
+    };
+
+    if (props.children) {
+      const children = (props.children.length
+        ? props.children
+        : [props.children]
+      ).filter(child => child.type === CtaMessage);
```

Comment:
This can likely be done in your prop type checking. Here's an article that has an example of a custom PropType validator that can validate the type of child being passed: http://www.mattzabriskie.com/blog/react-validating-children